src/components/frame/v5/pages/BalancePage/partials/BalanceTable/hooks.tsx

```diff
@@ -171,3 +234,137 @@ export const useGetTableMenuProps = () =>
       ],
     };
   }, []);
+
+export const useBalanceTable = (): UseFundsTableReturnType => {
+  const { colony } = useColonyContext();
+  const selectedTeam = useGetSelectedTeamFilter();
+  const { balances, nativeToken } = colony || {};
+  const [searchValue, setSearchValue] = useState('');
+  const [value, setValue] = useState<DeepPartial<BalanceTableFilters>>({
+    type: {},
+    attribute: {
+      native: false,
+      reputation: false,
+    },
+  });
+
+  const tokensData: TokensDataProps[] | undefined = useMemo(
+    () =>
+      colony?.tokens?.items.map((item) => {
+        const currentTokenBalance =
+          getBalanceForTokenAndDomain(
+            balances,
+            item?.token?.tokenAddress || '',
+            selectedTeam ? Number(selectedTeam.nativeId) : undefined,
+          ) || 0;
+        const decimals = getTokenDecimalsWithFallback(item?.token.decimals);
+        const convertedValue = convertToDecimal(
+          currentTokenBalance,
+          decimals || 0,
+        );
+
+        const formattedValue = getFormattedNumeralValue(
+          convertedValue,
+          currentTokenBalance,
+        );
+
+        return {
+          ...item,
+          balance: typeof formattedValue === 'string' ? formattedValue : '',
+        };
+      }),
+    [colony, balances, selectedTeam],
+  );
+
+  const sortedTokens = useMemo(
+    () =>
+      tokensData?.sort((a, b) => {
+        if (!a.balance || !b.balance) return 0;
+        return parseInt(b.balance, 10) - parseInt(a.balance, 10);
+      }),
+    [tokensData],
+  );
```

Comment:
This attempts to convert `balance` into a number, but balance will no longer always represent a number after it went through `getFormattedNumeralValue`. Something like 0.0000006 will be formatted into `6 x 10^-7` so the sorting will not work with it. You need to sort using the raw balance, and pass it through something like `BigNumber`, as you're dealing with numbers in excess of 10^18